### PR TITLE
[SB Pyamqp] stress updates

### PR DESCRIPTION
**Pull request by l0lawrence**

### CODE REVIEW

File: sdk/servicebus/azure-servicebus/stress/scripts/stress_test_base.py [hunk 1]
```diff
@@ -437,9 +436,9 @@ async def _receive_handle_message(self, message, receiver, end_time):
         )
 
     async def _receive_async(self, receiver, end_time):
-        _logger.critical("STARTING RECEIVER")
         self._schedule_interval_logger(end_time, "Receiver " + str(self))
         try:
+            _logger.critical("STARTING RECEIVER")
             async with receiver:
                 while end_time > datetime.utcnow() and not self._should_stop:
                     _logger.info("RECEIVE LOOP")
```
Comment (kashifkhan) [inline sdk/servicebus/azure-servicebus/stress/scripts/stress_test_base.py:444]: you want to change logging here as well ?

File: sdk/servicebus/azure-servicebus/stress/scenarios-matrix.yaml [hunk 2]
```diff
@@ -1,7 +1,32 @@
...
+    queuepullw:
+      testTarget: queuepullw
+    aqueuepullw:
+     testTarget: aqueuepullw
+    batchw:
+      testTarget: batchw
+    abatchw:
+      testTarget: abatchw
+    memray:
+      testTarget: memray
```
Comment (kashifkhan) [inline sdk/servicebus/azure-servicebus/stress/scenarios-matrix.yaml:30]: Thanks for putting in memray into the stress tests. I need to check with them if they have any suggestions on perhaps splitting the output file etc.

File: sdk/servicebus/azure-servicebus/stress/scripts/stress_test_base.py [hunk 3]
```diff
@@ -276,20 +288,29 @@ def _receive(self, receiver, end_time):
...
 
     def run(self):
+
         start_time = datetime.utcnow()
+        if isinstance(self.duration, int):
+            self.duration = timedelta(seconds=self.duration)
         end_time = start_time + (self._duration_override or self.duration)
+
         with self.process_monitor:
             with concurrent.futures.ThreadPoolExecutor(max_workers=4) as proc_pool:
```
Comment (kashifkhan) [inline sdk/servicebus/azure-servicebus/stress/scripts/stress_test_base.py:298]: how many senders and or receivers can there be? You could end up in a situation where there are 3 senders, 2 receivers and the last receiver never runs as it has to wait for a free thread.

Comment (l0lawrence) [reply sdk/servicebus/azure-servicebus/stress/scripts/stress_test_base.py:298]: all test right now are 1:1

**Review by swathipil — APPROVED** [review]
lgtm! thanks @l0lawrence!

**Review by kashifkhan — APPROVED** [review]

**Pull request merged**